HistoricalDataFetcher.Classes/Utilities/ApiRequest.cs

```diff
@@ -0,0 +1,172 @@
+﻿using HistoricalDataFetcher.Classes.DataLayer.Cache;
+using HistoricalDataFetcher.Classes.Models;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json;
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+using System.Timers;
+
+namespace HistoricalDataFetcher.Classes.Utilities
+{
+    public static class ApiRequest
+    {
+        private static string _username;
+        private static string _password;
+
+        private static string _httpHost;
+        private static string _urlBase;
+        private static ICache _cache;
+        private static Timer _timer;
+        private static AccessToken _accessTokenInstance;
+
+        public static string HttpHost
+        {
+            get
+            {
+                return _httpHost;
+            }
+        }
+
+        public static string UrlBase
+        {
+            get
+            {
+                return _urlBase;
+            }
+        }
+
+        /// <summary>
+        /// Sets the UrlBase and logs in to test the connection
+        /// </summary>
+        /// <param name="cache">Cache to use</param>
+        /// <param name="username">Username of the Metasys server</param>
+        /// <param name="password">Password of the Metasys server</param>
+        /// <param name="httpHost">Host of the Metasys server</param>
+        /// <returns></returns>
+        public static async Task<bool> InitializeAsync(ICache cache, string username, string password, string httpHost)
+        {
+            _cache = cache;
+            _username = username;
+            _password = password;
+            _httpHost = httpHost;
+            _urlBase = $"https://{_httpHost}/api";
```

Comment:
why not set it to /api/v1?

Reply:
version can be specified in route or header, omitting it will default to version 1. if we put /v1 in the our sample code, we will be encouraging users to use route to specify version. is this what we want to promote?

Reply:
We want to encourage them to be explicit.

Reply:
This has been changed to v1